src/option_menu.c

```diff
@@ -0,0 +1,929 @@
+#include "global.h"
+#include "gpu_regs.h"
+#include "scanline_effect.h"
+#include "text_window_graphics.h"
+#include "menu.h"
+#include "menu_helpers.h"
+#include "task.h"
+#include "palette.h"
+#include "overworld.h"
+#include "malloc.h"
+#include "help_system.h"
+#include "main.h"
+#include "bg.h"
+#include "window.h"
+#include "text.h"
+#include "text_window.h"
+#include "string_util.h"
+#include "strings.h"
+#include "field_fadetransition.h"
+#include "gba/m4a_internal.h"
+
+// Menu items
+enum
+{
+    MENUITEM_TEXTSPEED = 0,
+    MENUITEM_BATTLESCENE,
+    MENUITEM_BATTLESTYLE,
+    MENUITEM_SOUND,
+    MENUITEM_BUTTONMODE,
+    MENUITEM_FRAMETYPE,
+    MENUITEM_CANCEL,
+    MENUITEM_COUNT,
+};
+// Window Ids
+enum
+{
+    WIN_TEXT_OPTION,
+    WIN_OPTIONS,
+};
+
+static const struct WindowTemplate sOptionMenuWinTemplates[] =    //3CC2B8
```

Comment:
you don't need to add comments for offset in ROM, unless it helps people figure out NONMATCHING